index.js

```diff
@@ -325,20 +325,35 @@ class KeyringController extends EventEmitter {
   //
   // Attempts to sign the provided @object msgParams.
   // Prefixes the hash before signing as per the new geth behavior.
-  signPersonalMessage (msgParams) {
+  signPersonalMessage (msgParams, opts) {
     const address = normalizeAddress(msgParams.from)
     return this.getKeyringForAccount(address)
     .then((keyring) => {
-      return keyring.signPersonalMessage(address, msgParams.data)
+      return keyring.signPersonalMessage(address, msgParams.data, opts)
     })
   }
 
   // Sign Typed Message (EIP712 https://github.com/ethereum/EIPs/pull/712#issuecomment-329988454)
-  signTypedMessage (msgParams) {
+  signTypedMessage (msgParams, opts) {
     const address = normalizeAddress(msgParams.from)
     return this.getKeyringForAccount(address)
       .then((keyring) => {
-      return keyring.signTypedData(address, msgParams.data)
+      return keyring.signTypedData(address, msgParams.data, opts)
+    })
+  }
+
+  // returns an app key
+  getAppKeyAddress (_address, origin) {
```

Comment:
See here: https://github.com/MetaMask/eth-simple-keyring/pull/16#discussion_r316258931
The suggestions made in that comment apply here as well